app/src/main/java/org/simple/clinic/sync/LastSyncedState.kt

```diff
@@ -13,6 +13,15 @@ data class LastSyncedState(
     val lastSyncSucceededAt: Instant? = null
 ) {
 
+  fun withProgress(syncProgress: SyncProgress): LastSyncedState {
+    check(syncProgress != SUCCESS) { "Use success() instead if the sync is successful!" }
+    return this.copy(lastSyncProgress = syncProgress)
+  }
+
+  fun success(clock: UtcClock): LastSyncedState {
```

Comment:
Rename to `withProgressSuccess`?

Reply:
I added discrete generator functions for all sync progress events. Please take a look?